lib/src/kbasesearchengine/events/handler/WorkspaceEventHandler.java

```diff
@@ -570,4 +573,108 @@ private static String toWSRefPath(final List<GUID> objectRefPath) {
         return String.join(";", refpath);
     }
 
+    @Override
+    public StatusEvent updateEvent(
+            StatusEvent ev)
+            throws IndexingException,
+                   RetriableIndexingException {
+
+        // latest states to update
+        StatusEventType latestEventType = ev.getEventType();
+        Boolean latestIsPublic = null;
+        String latestName = null;
+
+        if( ev.isPublic().isPresent()) {
+            latestIsPublic = ev.isPublic().get().booleanValue();
+        }
+        if( ev.getNewName().isPresent()) {
+            latestName = ev.getNewName().get().toString();
+        }
+
+
+        // wsid and objid of the object that the specified StatusEvent is an event for
+        long wsid = Long.valueOf(ev.getAccessGroupId().get().toString()).longValue();
+        long objid = Long.valueOf(ev.getAccessGroupObjectId().get().toString()).longValue();
+
+        Map<String, Object> command;
+
+        // check if object is deleted and update state of latestEventType
+        command = new HashMap<>();
+        command.put("command", "listObjects");
+        command.put("params", new ListObjectsParams().withShowOnlyDeleted(wsid).withIds(Arrays.asList(wsid)));
+        final List<Tuple11> deletedObjList;
+        try {
+            deletedObjList = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(List.class);
+
+            for( Tuple11 obj: deletedObjList) {
+                long deletedObjid = Long.valueOf(obj.getE1().toString()).longValue();
+                if(deletedObjid  == objid) {
+                    latestEventType = StatusEventType.DELETED;
+                    break;
+                }
+            }
+        } catch (IOException e) {
+            throw new RetriableIndexingException(e.getMessage());
+        } catch (JsonClientException e) {
+            throw new RetriableIndexingException(e.getMessage());
+        }
+
+
+        // check if name has changed and update state of lastestName
+        ObjectSpecification os = new ObjectSpecification().withRef(
+                ev.getAccessGroupId().get().toString() + "/" +
+                        ev.getAccessGroupObjectId().get());
+
+        final List<ObjectSpecification> getInfoInput = new ArrayList<>();
+        getInfoInput.add(os);
+
+        command = new HashMap<>();
+        command.put("command", "getObjectInfo");
+        command.put("params", new GetObjectInfo3Params().withObjects(getInfoInput));
+
+        final GetObjectInfo3Results objInfo;
+        try {
+            objInfo = ws.getClient().administer(new UObject(command))
+                    .asClassInstance(GetObjectInfo3Results.class);
+            String name = objInfo.getInfos().get(0).getE2();
+            if ( !Utils.isNullOrEmpty(name) ) {
```

Comment:
I don't think there's any case where this can be false, is there?

Reply:
This was a paranoid check since I couldn't find anything in documentation that stated either way. I have removed the check.

Reply:
Yet another paranoid check which I have removed.